### PR TITLE
Add toast UI from radix

### PR DESCRIPTION
**Pull request by VLuisa**
**Describe changes:**
Added a toast component that dismisses after a delay or on a user action.

image

QA checklist:
- [x] Toast should be dismissed after default 5000 ms unless overidden
- [x] Consumer can customize visibility of close button and cta button
- [x] Toast should be vertically centered on screen and responsively shrink as viewport is narrowed
- [x] Documentation is clear and helpful

Reviewer: @tristen

### CODE REVIEW

File: src/components/toast/toast.test.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    afterEach(() => {
+      jest.useRealTimers();
+    });
+
+    test('fires onExit after duration', () => {
+      render(<Toast {...props}></Toast>);
+      expect(mockedOnExit).toHaveBeenCalledTimes(0);
+      jest.advanceTimersByTime(1001);
+      expect(mockedOnExit).toHaveBeenCalledTimes(1);
+    });
```
Comment (tristen) [inline src/components/toast/toast.test.tsx:68]: 🏄

File: src/components/toast/toast.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  let actionBtnClass = closeButton ? '' : 'pr12';
+  return (
+    <ToastPrimitive.Provider swipeDirection="down" duration={duration}>
+      <span
+        onClick={() => {
+          window.clearTimeout(timerRef.current);
+          timerRef.current = window.setTimeout(() => {}, 100);
+        }}
+      >
```
Comment (tristen) [inline src/components/toast/toast.tsx:38]: Is this still required?

File: src/components/toast/toast.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  /**
+   * A function called when popover is dismissed. You need to use this callback
+   * to remove the Toast from the rendered page.
+   */
+  onExit: PropTypes.func.isRequired,
+  /**
+   * Triggers the active state of the toast. When true, the toast appears.
+   */
+  active: PropTypes.bool.isRequired
+};
```
Comment (tristen) [inline src/components/toast/toast.tsx:120]: Another small nit-pick: can we move `isRequired` properties to the top of the list?

File: src/components/toast/toast.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  content: string;
+  action?: {
+    text: string;
+    callback: () => void;
+  };
+  duration?: number;
+  closeButton?: boolean;
+  children: ReactNode;
+  active: boolean;
+  onExit: () => void;
```
Comment (tristen) [inline src/components/toast/toast.tsx:16]: Small nit-picky point but to match convention: can we move required fields about optional ones?

**Review by tristen — CHANGES_REQUESTED** [review]
Small stuff. Otherwise solid work! 🍞 🥂

File: src/components/toast/toast.tsx (hunk 2 above)
Comment (VLuisa) [reply src/components/toast/toast.tsx:38]: ah good catch, it is not!

File: src/components/toast/toast.tsx (hunk 3 above)
Comment (VLuisa) [reply src/components/toast/toast.tsx]: oh cool, TIL!

**Review by tristen — APPROVED** [review]

**Pull request merged**